### PR TITLE
Parameterize the l10n download link tests

### PR DESCRIPTION
**Pull request by davehunt**
This builds on pull #438 so should be held until that's merged. It removes existing duplication (tests/test_l10n_download_links.py and tests/test_localization.py appear to check the same things) but also refactors the tests to be parameterised. This allows for better use of resources as we can run the individual tests in parallel, and provides a better representation of coverage and failures. It also means that when running these tests locally you get regular feedback, rather than individual tests running for several minutes without reporting anything to the console.

### CODE REVIEW

**Comment by davehunt** [comment]
Adhoc: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/mozorg.adhoc/32/

Before: 122 tests ran in 151 seconds
After: 340 tests ran in 100 seconds

**Comment by davehunt** [comment]
@bobsilverberg would you mind taking a look at this and giving feedback?

File: tests/test_l10n_download_links.py [hunk 1]
```diff
@@ -7,74 +6,47 @@
 import pytest
 import requests
 
+pytestmark = [pytest.mark.nondestructive, pytest.mark.skip_selenium]
```
Comment (bobsilverberg) [inline tests/test_l10n_download_links.py]: Nice. I didn't know about `pytestmark`.

File: tests/test_l10n_download_links.py [hunk 2]
```diff
@@ -7,74 +6,47 @@
...
+    r = requests.get('{0}{1}'.format(base_url, page))
+    soup = BeautifulSoup(r.content, 'html.parser')
+    table = soup.find('table', class_='build-table')  # find the table of links
+    idlist = []
+    argvalues = []
+    for row in table.find('tbody').find_all('tr'):
+        locale = row['id']
+        idlist.append(locale)  # the test id should be the locale
+        url = row.find('a')['href']
+        argvalues.append([locale, url])  # pass locale and url to each test
```
Comment (bobsilverberg) [inline tests/test_l10n_download_links.py]: Uber-nit: According to the docs [1] this should be a tuple, not a list.

[1] https://pytest.org/latest/parametrize.html#pytest-generate-tests

File: tests/test_l10n_download_links.py [hunk 3]
```diff
@@ -7,74 +6,47 @@
...
-        '''
-        url = "%s/%s" % (mozwebqa.base_url, link)
-        page_response = requests.get(url)
-        html = BeautifulSoup(page_response.content)
-        language_rows = html.findAll('tr', id=True)
-        return language_rows
+def check_link(locale, url):
+    r = requests.head(url, allow_redirects=True)
+    assert locale in r.url  # the correct locale is in the final url
+    assert requests.codes.found == r.history[0].status_code  # url redirects
```
Comment (bobsilverberg) [inline tests/test_l10n_download_links.py]: The old code allows for both 301 and 302. Is this new code okay, or should we be allowing for both of those still?

**Comment by bobsilverberg** [comment]
Very nice, @davehunt, I like it!  r+ wc

File: tests/test_l10n_download_links.py (hunk 1 above)
Comment (jgmize) [inline tests/test_l10n_download_links.py]: Neither did I. For reference, here are the docs: https://pytest.org/latest/example/markers.html#marking-whole-classes-or-modules

**Comment by davehunt** [comment]
This was merged in https://github.com/mozilla/mcom-tests/commit/2e607646cb81a5c2561b331e15caccafe7231963

**Pull request closed without merging**